src/instruction.cpp

```diff
@@ -124,7 +122,7 @@ bool instruction::valid() const
     }
     else if(op.name() == "@return")
     {
-        computed = {};
+        computed = get_shape();
```

Comment:
I think this whole `if` branch can be removed.

Reply:
Done.